### ISSUE TITLE
BrushNet fails to clone model before patching, generating "BrushNet inference: there is no brushnet_model in transformer_options"

### ISSUE THREAD

**Issue by ttulttul** [nullquant/ComfyUI-BrushNet#49]
Edit: what I wrote below was completely false. See PR#50. The model must be cloned before patching.

The model patcher checks for the model type, but if the model type isn't SDXL or SD1.5, then patching effectively fails without any error to the user:

https://github.com/nullquant/ComfyUI-BrushNet/blob/bd7c521b7b1e729223d667ca77148667873372bc/brushnet_nodes.py#L470

**Comment by ttulttul** [comment]
Please see https://github.com/nullquant/ComfyUI-BrushNet/pull/50

**Comment by ttulttul** [comment]
BTW - great work. This was a lot of work to pull together a native ComfyUI node. And it's the best inpaint I have found.

**Comment by nullquant** [comment]
Thank you for good words :). Thank you for a commit, but I already tried to clone model in the code - see line 289 of brushnet_nodes.py - I cloned model just before patch. But it is no use. 

I patched diffusion model and all clones of ModelPatcher have it. So even after cloning you still have this message. If we deep clone diffusion model the needed VRAM will be doubled. Some people already have VRAM problems so it is undesirable.

**Comment by nullquant** [comment]
It is a simple diagnostic message. We can comment it out, or we can write a special class for cloning the diffusion model without weights, and have a lot of probable bugs and undocumented behavior along the way. :) I think it is ok as it is.

**Comment by ttulttul** [comment]
Weird. This patch eliminated the problem on my end stall.

**Comment by ttulttul** [comment]
I see what you mean now. The `clone()` method does not clone `model.model`. I wonder if there is an established way of doing this for model patchers that require modifying the forward pass, without cloning the model weights?

**Comment by nullquant** [comment]
As far as I know - no. I've checked several other repos, most of the time people patch `forward_timestep_embed` in 'ComfyUI/comfy/ldm/modules/diffusionmodules/openaimodel.py'. It is a code patch so it will stay till ComfyUI restart. I started from this too. Now I patch `model.model.diffusion_model` object instance, so at least we can reload model. 

There was a discussion on reddit about `diffusers` and ComfyUI uncompatibility: https://www.reddit.com/r/comfyui/comments/17fvb49/whats_up_with_diffusers_and_comfyui/ and ComfyUI's author (comfyanonymous) said:
"...The diffusers library itself is badly designed so I won't include or use it in the core software but I don't mind supporting models in that format as long as it's something people are actually going to use."

There is a lot of AI research groups that publish new models every week and most of them using `diffusers` library. So I would love to use some established way, but find none.
